### PR TITLE
Introducing metrics-server

### PR DESCRIPTION
**Pull request by piosz**
ref https://github.com/kubernetes/features/issues/271

There is still some work blocked on problems with repo synchronization:
- migrate to `v1beta1` introduced in #51653 
- bump deps to HEAD
Will do it in a follow up PRs once the issue is resolved.

```release-note
Introduced Metrics Server
```

### CODE REVIEW

File: cluster/addons/metrics-server/auth-delegator.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+kind: ClusterRoleBinding
+metadata:
+  name: metrics-server:system:auth-delegator
+  labels:
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:auth-delegator
```
Comment (mikedanese) [inline cluster/addons/metrics-server/auth-delegator.yaml]: this looks find but cc @deads2k

**Comment by piosz** [comment]
removed lgtm label to let @DirectXMan12 to take a look

File: cluster/addons/metrics-server/auth-delegator.yaml (hunk 1 above)
Comment (piosz) [reply cluster/addons/metrics-server/auth-delegator.yaml]: this was reviewed in https://github.com/kubernetes-incubator/metrics-server/pull/4

Comment (deads2k) [reply cluster/addons/metrics-server/auth-delegator.yaml]: > this looks find but cc @deads2k

That's the role name. It allows the caller to hit tokenaccessreviews and subjectaccessreviews.

**Comment by DirectXMan12** [comment]
looks fine at a glance.  Will take a closer look a bit later if need be.  metrics-server needs to be updated to use metrics.k8s.io/v1beta1, otherwise it won't work once the resource metrics API update merges, though.

**Comment by piosz** [comment]
@DirectXMan12 will update this once #51653 is merged. It's currently not used by HPA so it's safe to merge it.

File: cluster/addons/metrics-server/metrics-server-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        image: gcr.io/google_containers/metrics-server-amd64:dev
+        imagePullPolicy: Always
+        # TODO(piosz): revisit resources
+        resources:
+          requests:
+            memory: 100Mi
+        command:
+        - /metrics-server
+        - --source=kubernetes.summary_api:''
+      tolerations:
```
Comment (DirectXMan12) [inline cluster/addons/metrics-server/metrics-server-deployment.yaml]: Please list ports here

File: cluster/gce/config-default.sh [hunk 3]
```diff
@@ -123,6 +123,12 @@ ENABLE_L7_LOADBALANCING="${KUBE_ENABLE_L7_LOADBALANCING:-glbc}"
 #   standalone     - Heapster only. Metrics available via Heapster REST API.
 ENABLE_CLUSTER_MONITORING="${KUBE_ENABLE_CLUSTER_MONITORING:-influxdb}"
 
+# Optional: Enable Metrics Server. Metrics Server should be enable everywhere,
+# since it's a critical component, but in the first release we need a way to disable
+# this in case of stability issues.
+# TODO(piosz) remove this option once Metrics Server became a stable thing.
+ENABLE_METRICS_SERVER="${KUBE_ENABLE_METRICS_SERVER:-true}"
+
```
Comment (DirectXMan12) [inline cluster/gce/config-default.sh]: If this is going to be enabled by default, we should be using it for something, and should not deploy (or at least warn about) Heapster.  Turn it on in a separate PR once we're ready to make the HPA use it by default.

**Review by DirectXMan12 — CHANGES_REQUESTED** [review]
Squash RBAC fix, plus comments inline

File: cluster/gce/config-default.sh (hunk 3 above)
Comment (piosz) [reply cluster/gce/config-default.sh]: I'd like  to switch HPA in 1.8 to use metrics server.

Heapster is used by dashboard. Also we should perform a graceful switch and have both metrics-server and Heapster enabled in one release (1.8).

File: cluster/addons/metrics-server/metrics-server-deployment.yaml (hunk 2 above)
Comment (piosz) [reply cluster/addons/metrics-server/metrics-server-deployment.yaml]: done

**Comment by piosz** [comment]
@DirectXMan12 comments addressed. PTAL

File: cluster/gce/config-default.sh (hunk 3 above)
Comment (DirectXMan12) [reply cluster/gce/config-default.sh]: Ah, ack.

**Review by DirectXMan12 — APPROVED** [review]

**Comment by jdumars** [comment]
@piosz could you please add a SIG label?  Thanks!

**Pull request merged**